tf-modules/archive/iam.tf

```
@@ -186,10 +186,173 @@ data "aws_iam_policy_document" "lambda_api_gateway_policy" {
+  role       = aws_iam_role.ecs_execution_role.name
+  policy_arn = "arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryFullAccess"
+}
+
+resource "aws_iam_role_policy_attachment" "cloudwatch-task-policy-attach" {
+  role       = aws_iam_role.ecs_execution_role.name
+  policy_arn = "arn:aws:iam::aws:policy/CloudWatchFullAccess"
+}
+
+# ECS task role
+
+data "aws_iam_policy_document" "ecs_task_assume_role_policy" {
+  statement {
+    principals {
+      type        = "Service"
+      identifiers = ["ecs-tasks.amazonaws.com"]
+    }
+    actions = ["sts:AssumeRole"]
+  }
+}
+
+data "aws_iam_policy_document" "ecs_task_role_policy" {
+  statement {
+    actions = [
+      "lambda:GetFunction",
+      "lambda:invokeFunction"
+    ]
+    resources = ["*"]
+  }
+
+    statement {
+    actions = [
+      "s3:GetAccelerateConfiguration",
+      "s3:GetBucket*",
+      "s3:GetLifecycleConfiguration",
+      "s3:GetReplicationConfiguration",
+      "s3:ListBucket*",
+      "s3:PutAccelerateConfiguration",
+      "s3:PutBucket*",
+      "s3:PutLifecycleConfiguration",
+      "s3:PutReplicationConfiguration"
+    ]
+    resources = [for b in local.all_bucket_names : "arn:aws:s3:::${b}"]
+  }
+
+  statement {
+    actions = [
+      "s3:AbortMultipartUpload",
+      "s3:DeleteObject",
+      "s3:DeleteObjectVersion",
+      "s3:GetObject*",
+      "s3:ListMultipartUploadParts",
+      "s3:PutObject*"
+    ]
+    resources = [for b in local.all_bucket_names : "arn:aws:s3:::${b}/*"]
+  }
+
+  statement {
+    actions   = [
+      "dynamodb:DeleteItem",
+      "dynamodb:GetItem",
+      "dynamodb:PutItem",
+      "dynamodb:UpdateItem",
+      "dynamodb:Scan",
+      "dynamodb:Query"
+    ]
+    resources = [for k, v in var.dynamo_tables : v.arn]
+  }
+
+  statement {
+    actions   = ["dynamodb:Query"]
+    resources = [for k, v in var.dynamo_tables : "${v.arn}/index/*"]
+  }
+
+  statement {
+    actions = [
+      "states:DescribeActivity",
+      "states:DescribeExecution",
+      "states:GetActivityTask",
+      "states:GetExecutionHistory",
+      "states:SendTaskFailure",
+      "states:SendTaskSuccess"
+    ]
+    resources = ["arn:aws:states:*:*:*"]
+  }
+
+  statement {
+    actions = [
+      "kinesis:describeStream",
+      "kinesis:ListShards",
+      "kinesis:getShardIterator",
+      "kinesis:GetRecords"
+    ]
+    resources = ["arn:aws:kinesis:*:*:*"]
+  }
+
+  statement {
+    actions = [
+      "sqs:Send*",
+      "sqs:GetQueueUrl",
+    ]
+    resources = ["arn:aws:sqs:*:*:*"]
+  }
+
+  statement {
+    actions = ["secretsmanager:GetSecretValue"]
+    resources = [
+      aws_secretsmanager_secret.api_cmr_password.arn,
+      aws_secretsmanager_secret.api_launchpad_passphrase.arn,
+      var.rds_user_access_secret_arn
+    ]
+  }
+
+  statement {
+    actions = [
+      "sns:publish",
+      "sns:Subscribe",
+      "sns:Unsubscribe",
+      "sns:List*",
+    ]
+    resources = ["*"]
```

```suggestion
    actions = ["sns:Publish"]
    resources = [
      aws_sns_topic.report_executions_topic.arn,
      aws_sns_topic.report_granules_topic.arn,
      aws_sns_topic.report_pdrs_topic.arn
    ]
```

I know I made these changes, but I think we should be more restrictive and only give the permissions that we need, which should be these

---

Is this an update we should make in a separate PR?      Making non-merge conflict related updates in a feature branch merge doesn't feel great.    Disregard if this is missing the premise of the comment.

---

In this case, I **think** it's the right choice. So when I updated `master` to use Fargate for async operations, we had to create a new IAM role giving that Fargate task the necessary permissions.

However, on this RDS phase 2 branch, we have also updated our database write code so that publishing to SNS happens as part of the database write logic, not in a separate lambda. So Fargate tasks that are updating granules to "queued" or otherwise writing records need these SNS permissions whereas they didn't need those permissions in `master`. 

And we weren't aware this would be a problem until now when we're merging `master` into the `feature/rds-phase-2` branch, so this feels like the appropriate place to address it. Does that make sense?

---

Ah, I see.   Thanks for clarifying.